### DISCUSSION TITLE
Ideas for supercharging Command Search

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1172] [Ideas]
Command Search is a great way to quickly find commands that would take a lot longer locate in menus and preferences and help files.  I use CS daily and, because of it, work mostly in full screen mode, knowing that features are only a Ctrl+F1 away.

During my use, however, I have caught myself thinking about enhancements that would supercharge Command Search ("Feature Request" is my middle name).

I'd like to see if these ideas resonate with anyone else before I pester Dimitry with actual feature requests.  Hopefully your comments will clarify and enhance (and maybe shoot down) my ideas and improve my final feature request(s).

Mostly, these ideas related to Menu Items.

I wish there were additional keys for navigation through the population of items, such as page up/down, End, Home.

Where appropriate, I wish CS showed you the current status of features.  For example, "Navigate > Fold > Always unfolded node (set/reset)" -  is it currently checked?  That would better inform my decision to pull the trigger in CS.  (Of course, that would add to the CS width, taking up even more screen real estate.)

I wish CS had an option to open a menu cascade the same way CS takes you into Preferences to specific settings (perhaps initiated by Ctrl+click or Ctrl+Enter).  That would make it easier to assign keyboard shortcuts, which are created via the menus.

Or, even better, if you could right-click on CS items for a context menu:  status, assign hotkey, open menu  ...

Some of these requests might make sense if only a single node were selected vs multiple selected nodes.

None of them make sense if it meant taking a lot of developer time and effort.

Already CS is my best friend in Freeplane.  It would be great if it could be my all-around personal assistant.

**Comment by dpolivaev** [comment]
Control click on menu item in the cs dialog allows you to assign hotkeys

**Reply by kenfhill84083** [reply]
Ctrl double-click works for me.  Wow.  Freeplane is so comprehensive and robust you daily learn something new about it.  Thanks.  (I wish I'd known this before I made a Youtube video about Command Search a few days ago.)
